dev/src/index.ts

```diff
@@ -580,14 +580,21 @@ export class Firestore implements firestore.Firestore {
           const grpcModule = this._settings.grpc ?? require('google-gax').grpc;
           const sslCreds = grpcModule.credentials.createInsecure();
 
-          client = new module.exports.v1(
-            {
-              sslCreds,
-              ...this._settings,
-              fallback: useFallback,
-            },
-            gax
-          );
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const settings: any = {
```

Comment:
`settings` probably has type `ClientOptions | firestore.Settings`. The `ClientOptions` type comes from `google-gax`, it can be added to the import on line 19.

Reply:
Thanks for calling this out. ClientOptions is indeed the correct type for this constructor.